core/src/backend/navigator.rs

```diff
@@ -11,6 +12,39 @@ use std::sync::mpsc::{channel, Receiver, Sender};
 use std::task::{Context, Poll, RawWaker, RawWakerVTable, Waker};
 use std::time::Duration;
 use swf::avm1::types::SendVarsMethod;
+use url::{ParseError, Url};
+
+/// Attempt to convert a relative filesystem path into an absolute `file:///`
+/// URL.
+///
+/// If the relative path is an absolute path, the base will not be used, but it
+/// will still be parsed into a `Url`.
+#[cfg(any(unix, windows, target_os = "redox"))]
+pub fn url_from_relative_path<P: AsRef<Path>>(base: P, relative: &str) -> Result<Url, ParseError> {
+    let parsed = Url::from_file_path(relative);
+    if let Err(()) = parsed {
+        let base =
+            Url::from_directory_path(base).map_err(|_| ParseError::RelativeUrlWithoutBase)?;
+
+        return base.join(relative);
+    }
+
+    Ok(parsed.unwrap())
+}
```

Comment:
Can we add the `#[cfg(not(...))]` variant here that just returns an error or returns the URL as-is, and then we can remove the conditionals that are elsewhere?